test/e2e/persistent_volumes.go

```diff
@@ -32,6 +32,21 @@ import (
 	"k8s.io/kubernetes/test/e2e/framework"
 )
 
+// Data type to track PVs as keys and their bound PVCs as values.  All PVs should be stored in the map regardless of their status.
+// Unbound PVs keys have nil values. PVCs are mutually exclusive between the pvclist and the bindmap. One or the other
+// will track a PVC, but never both at once.  Once a PV is identified as bound, the respective PVC should be assigned to that
+// PV's value and set to nil in the pvclist object
+// NOTE: When working with PV keys, it is safe to delete them from the map using the delete() built-in.  However, you cannot
+// update the key itself.  For example, if you Get() a new instance of a PV, you must delete original PV key while adding the
+// new version to the map.  WARNING: It's unsafe to add keys to a map in a loop.  Their insertion in the map is unpredictable
+// and can result in the same pv.Name being iterated over again.
+type bindmap map[*api.PersistentVolume]*api.PersistentVolumeClaim
+
+// Data type to store PVCs prior to testing their status.  As noted above, when a PV status becomes "Bound" and the bound PVC identified,
+// assign the PVC as the value of the PV and set to nil in the pvclist.  It is important to maintain the mutual exclusivity of each PVC
+// between the bindmap and the pvclist be.  When removing a PVC, the element should be nil-ed.
+type pvclist []*api.PersistentVolumeClaim
```

Comment:
I should have read this comment before writing the previous one.  You're right, I think using 2 maps of `string(name):bool(isbound)`, then Get()ing the objects as needed is cleaner and makes their purpose a bit more intuitive.